spv/sync.go

```diff
@@ -70,10 +71,24 @@ type Syncer struct {
 	currentLocators   []*chainhash.Hash
 	locatorGeneration uint
 	locatorMu         sync.Mutex
+
+	// Holds all of the potential callbacks to update the grpc request
+	notificationCallbacks *NtfnsCallbacks
+}
+
+// NtfnsCallbacks struct to contain all of the upcoming callbacks that will
+// be used to update the rpc streams for syncing.
+type NtfnsCallbacks struct {
+	// Update when the wallet becomes synced or unsynced
+	SyncUpdated SyncedUpdated
 }
 
+// SyncedUpdated is defined for a callback to notify when the wallet is seen
+// to be synced or unsynced from its connected peers.
+type SyncedUpdated func(sync bool)
+
 // NewSyncer creates a Syncer that will sync the wallet using SPV.
-func NewSyncer(w *wallet.Wallet, lp *p2p.LocalPeer) *Syncer {
+func NewSyncer(w *wallet.Wallet, lp *p2p.LocalPeer, callbacks *NtfnsCallbacks) *Syncer {
```

Comment:
Instead of changing this API, please add a new method which sets the callbacks, like how the persistent peers are set.